stalePullRequest.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+var Cesium = require('cesium');
+var Promise = require('bluebird');
+var requestPromise = require('request-promise');
+
+var defined = Cesium.defined;
+
+var dateLog = require('./lib/dateLog');
+var checkStatus = require('./lib/checkStatus');
+var Settings = require('./lib/Settings');
+
+module.exports = stalePullRequest;
+
+if (require.main === module) {
+    Settings.loadRepositoriesSettings('./config.json')
+        .then(stalePullRequest)
+        .catch(function (err) {
+            dateLog('Received error: ' + err);
+        });
+}
+/** Bump stale pull requests for each repository
+ *
+ * @param {String[]} repositoryNames Names of repositories
+ * @returns {Promise<Array<http.IncomingMessage | undefined> | undefined>} Promise to an array of incoming messages
+ */
+function stalePullRequest(repositoryNames) {
+    return Promise.all(
+        repositoryNames.map(function (repositoryName) {
+            var bumpStalePullRequests = Settings.repositories[repositoryName].bumpStalePullRequests;
+            if (!defined(bumpStalePullRequests)) {
+                dateLog('Repository ' + repositoryName + ' does not have `bumpStalePullRequests` turned on');
+                return Promise.resolve();
+            }
+            return stalePullRequest.implementation(bumpStalePullRequests.url + '?sort=updated&direction=asc',
+                Settings.repositories[repositoryName].gitHubToken, Settings.repositories[repositoryName].maxDaysSinceUpdate);
+        })
+    );
+}
+
+/** Implementation
+ *
+ * @param {String} pullRequestsUrl Base url to list pull requests https://developer.github.com/v3/pulls/#list-pull-requests
+ * @param {String} gitHubToken Token to verify with github
+ * @param {Number} maxDaysSinceUpdate Maximum days since a PR has been updated
+ * @return {Promise<Array<http.IncomingMessage | undefined>>} Promise to an array of incoming messages
+ */
+stalePullRequest.implementation = function (pullRequestsUrl, gitHubToken, maxDaysSinceUpdate) {
+    var headers = {
+        'User-Agent': 'cesium-concierge',
+        Authorization: 'token ' + gitHubToken
+    };
+    return requestPromise.get({
+        uri: pullRequestsUrl,
+        headers: headers,
+        json: true,
+        resolveWithFullResponse: true
+    })
+    .then(function (pullRequestsJsonResponse) {
+        return checkStatus(pullRequestsJsonResponse);
+    })
+    .then(function (pullRequestsJsonResponse) {
+        var firstMessage = 'It looks like this pull request hasn\'t been updated in a while!\n\n' +
+            'Please update it soon or close it!';
+        var alreadyBumpedMessage = 'Hi again!\n\nThis pull request has not been active in a while. Please consider ' +
+            'closing it!';
+        return Promise.all(pullRequestsJsonResponse.body.map(function(pullRequest) {
```

Comment:
There is a Promise.map you can use here instead, but I would rather use `Promise.each(pullRequestsJsonResponse.body.map, function(pullRequest) {` here, or you might end up issuing too many REST calls at once (because they could all be in flight at the same time).